### PR TITLE
Remove userzone and remove all federation config

### PR DESCRIPTION
**Pull request by devincowan**
Resolves https://github.com/hydroshare/hydroshare/issues/5519 (by removing the userzone and all federation)
Resolves https://github.com/hydroshare/hydroshare/issues/5512 (by adding a GH action for PR builds) -- rebased thanks to @sblack-usu 
Resolves https://github.com/hydroshare/hydroshare/issues/5104 (because no more UserZone accounts)
Resolves https://github.com/hydroshare/hydroshare/issues/5061 (because messaging is updated)
Resolves https://github.com/hydroshare/hydroshare/issues/3915 (because no more copying bags to userzone)
Resolves https://github.com/hydroshare/hydroshare/issues/1930 (because we are moving away from irods federation)
Resolves https://github.com/hydroshare/hydroshare/issues/2585 (because no more irods passwords)

### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. [Enter positive test case here]

### CODE REVIEW

File: hs_core/views/utils.py [hunk 1]
```diff
@@ -329,6 +329,7 @@ def run_ssh_command(host, uname, exec_cmd, pwd=None, private_key_file=None):
     return output
 
 
+# TODO: can we remove this?
```
Comment (devincowan) [inline hs_core/views/utils.py:332]: unsure what HYRAX is...I'm wondering if we can get rid of this

Comment (sblack-usu) [reply hs_core/views/utils.py:332]: hyrax == thredds
tds-proxy.hydroshare.org

Comment (devincowan) [reply hs_core/views/utils.py]: Thank you!

File: irods_browser_app/views.py [hunk 2]
```diff
@@ -123,24 +123,18 @@ def upload_add(request):
         return JsonResponse({'error': "Invalid file type: {ext}".format(ext=ext)},
                             status=status.HTTP_400_BAD_REQUEST)
     else:
-        homepath = irods_fnames_list[0]
-        # TODO: this should happen whether resource is federated or not
-        irods_federated = utils.is_federated(homepath)
```
Comment (devincowan) [inline irods_browser_app/views.py:128]: just making a note for memory that removing this is_federated call should speed up the latency we have been seeing with users.hysroshare.org -> cloud

File: irods/README.md [hunk 3]
```diff
@@ -27,10 +27,10 @@ $ ./use-local-irods.sh
   $ cd irods
   $ ./use-local-irods.sh --persist
   ```
-  The `--persist` flag will create two new directories as `/home/${USER}/icat1` and `/home/${USER}/icat2` where iRODS vault and database files will be persisted. These directories will remain in place until the user manually destroys them.
+  The `--persist` flag will create a new directory as `/home/${USER}/icat1` where iRODS vault and database files will be persisted. These directories will remain in place until the user manually destroys them.
 
-The script will run, deploy and federate two iCAT servers, and modify the three aforementioned files to be configured to use the 
-newly created iRODS Docker servers. Once the script has completed, return back the the main `hydroshare` directory and run
+The script will run, deploy and federate an iCAT server, and modify the three aforementioned files to be configured to use the
```
Comment (sblack-usu) [inline irods/README.md:32]: how will it federate an iCAT server when there is only one server?

File: irods_log.log [hunk 4]
```diff
@@ -0,0 +1,161 @@
+[02/Jul/2024 14:18:30] INFO [irods_browser_app.views:114] ************************************************************************************************************************************************************************************************************************************************************************************************************
```
Comment (sblack-usu) [inline irods_log.log:1]: This looks like it was accidentally included in a commit

File: hs_core/hydroshare/utils.py [hunk 5]
```diff
@@ -238,6 +186,7 @@ def get_federated_zone_home_path(filepath):
     Returns:
         the zone name extracted from filepath
     """
+    # TODO: perhaps add HS_IRODS_PROXY_USER_IN_USER_ZONE back in
```
Comment (sblack-usu) [inline hs_core/hydroshare/utils.py]: I think this comment isn't needed anymore?

File: hs_core/tests/api/native/test_add_resource_files.py [hunk 6]
```diff
@@ -100,7 +100,7 @@ def test_add_files_over_quota(self):
 
         uquota = self.user.quotas.first()
         # make user's quota over hard limit 125%
-        uquota.user_zone_value = uquota.allocated_value * 1.3
```
Comment (sblack-usu) [inline hs_core/tests/api/native/test_add_resource_files.py:103]: Missed an import https://github.com/hydroshare/hydroshare/blob/manual-irods-userzone-and-rm-federation/hs_core/tests/api/native/__init__.py#L19

File: hs_core/migrations/0083_auto_20240702_2038.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+            model_name='baseresource',
+            name='resource_federation_path',
+        ),
+        migrations.AlterIndexTogether(
+            name='resourcefile',
+            index_together={('object_id', 'resource_file')},
+        ),
+        migrations.RemoveField(
+            model_name='resourcefile',
+            name='fed_resource_file',
```
Comment (sblack-usu) [inline hs_core/migrations/0083_auto_20240702_2038.py:23]: So satisfying to se this field go!

**Comment by sblack-usu** [comment]
Looks like only the pytest tests ran http://ci.hydroshare.org:8080/job/hydroshare-pull-requests/7716/testReport/

File: theme/static/js/profile.js [hunk 8]
```diff
@@ -172,64 +172,10 @@ function irods_status_info(alert_type, status, title) {
...
-        },
-        error: function(xhr, errmsg, err) {
-            $('#delete-irods-account-dialog').modal('hide');
-            var irodsContainer = $("#irods-account-container");
-            const errorText = JSON.parse(xhr.responseText).error
-            errorText && irodsContainer.append(irods_status_info('alert-warning', errorText, 'Failure'));
-        }
-    });
-    return false;
+    let email = 'help@cuahsi.org';
```
Comment (sblack-usu) [inline theme/static/js/profile.js:175]: What is the strategy you are thinking for this? Is this for access to the renci userzone before it goes away?

**Comment by sblack-usu** [comment]
> Looks like only the pytest tests ran http://ci.hydroshare.org:8080/job/hydroshare-pull-requests/7716/testReport/

Removing the bad import fixes this issue and runs all the tests.

File: theme/static/js/profile.js (hunk 8 above)
Comment (sblack-usu) [reply theme/static/js/profile.js:175]: I'm recalling this in one of our meetings discussing userzone now.

File: hs_core/migrations/0083_auto_20240702_2038.py (hunk 7 above)
Comment (devincowan) [reply hs_core/migrations/0083_auto_20240702_2038.py:23]: AGREED!

File: theme/migrations/0036_auto_20240702_1847.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+    dependencies = [
+        ('theme', '0035_remove_userquota_data_zone_value'),
+    ]
+
+    operations = [
+        migrations.RemoveField(
+            model_name='userquota',
+            name='user_zone_value',
+        ),
+        migrations.AlterField(
```
[the remark this answers is not in the document]
Comment (devincowan) [reply theme/migrations/0036_auto_20240702_1847.py:17]: sorry for the confusion. maybe we will squash migrations

File: irods/README.md (hunk 3 above)
Comment (devincowan) [reply irods/README.md:32]: good call, it wont! :)

File: irods_log.log (hunk 4 above)
Comment (devincowan) [reply irods_log.log]: yep, thanks

File: hs_core/hydroshare/utils.py (hunk 5 above)
Comment (devincowan) [reply hs_core/hydroshare/utils.py]: yes, thanks removed it

File: hs_core/tests/api/native/test_add_resource_files.py (hunk 6 above)
Comment (devincowan) [reply hs_core/tests/api/native/test_add_resource_files.py:103]: interesting flake8 didn't catch that. Nice eye!

File: theme/migrations/0036_auto_20240702_1847.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (sblack-usu) [reply theme/migrations/0036_auto_20240702_1847.py:17]: I don't think it is worth squashing the merges, this is fine

**Comment by devincowan** [comment]
Failure of the Jenkins PR is expected (because I removed `use-local-irods.sh`). The GH actions tests are accurate and passing

File: hs_core/hydroshare/hs_requests.py [hunk 10]
```diff
@@ -14,17 +14,12 @@
 from django.conf import settings
 
 
-def get_nginx_ip():
+def get_local_ip():
     """
     Read the Nginx IP address from a local file, where it is stored during
     every nginx restart. This address is dynamic and changes on every restart.
```
Comment (sblack-usu) [inline hs_core/hydroshare/hs_requests.py:20]: This message is no longer true

**Review by sblack-usu — APPROVED** [review]

**Pull request merged**